### DISCUSSION TITLE
Socket.io Stuck in long pulling on production

### DISCUSSION THREAD

**Discussion by obipascal** [socketio/socket.io#4035] [Q&A] [answered]
I have a socket.io server that accepts connection and i have deployed it for production use and it's working well except that it is stuck in long pulling and never upgrade to websocket connection i have tried the solution as specified [here](https://socket.io/docs/v4/troubleshooting-connection-issues/#Problem-the-socket-is-stuck-in-HTTP-long-polling) yet is still stuck in long pulling, i also use the https and supplied my key and ca yet when the connection is upgrading to wss://messenger.watunet.com the connection never upgrades at all and the status for this request on chrome browser is finished.

Please is they a way i can resolve this issue?

**Comment by darrachequesne** [comment, marked as answer]
In most cases it's a misconfiguration of the reverse proxy in front of the server. Did you check https://socket.io/docs/v4/reverse-proxy/ ?

Besides, it seems I can't access the server at the URL you provided:

```
$ curl "https://messenger.watunet.com/socket.io/?EIO=4&transport=polling"
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>Cannot GET /socket.io/</pre>
</body>
</html>
```

**Reply by obipascal** [reply]
Okay in other to access the server  add a path messenger 

`https://socket.watunet.com/messenger/socket.io/?EIO=4&transport=polling`

**Reply by darrachequesne** [reply]
Yes, the `path` option must indeed be the same on the client and the server:

*Server*

```js
const io = new Server(httpServer, {
  path: "/my-custom-path/"
});
```

*Client*

```js
import { io } from "socket.io-client";

const socket = io("https://example.com", {
  path: "/my-custom-path/"
});
```

Reference: https://socket.io/docs/v4/server-options/#path

**Reply by obipascal** [reply]
Sorry the not path actually its a namespace rather that's why there is a forward slash, You are right about the server wrong configuration because the same code was used on a different server and it upgraded to wss://socket.obitechinvent.com/payve and connected successfully. So i think is a server miss config but then i don't know how to configure the server to be able to work well as descripted in the docs for proxied reserve servers am running Centos 7 Linux servers
